### PR TITLE
Add azure mongo driver

### PR DESCRIPTION
**Pull request by vachillo**
Adding support for Azure CosmosDB MongoDB vCore API.
- making `index_name` a required field (its not technically required for the azure driver but its there anyway)
- adding required field `vector_path` to specify the field name of the vector content in the docs in the db

### CODE REVIEW

**Review by cjkindel — APPROVED** [review]

File: griptape/drivers/vector/azure_mongodb_vector_store_driver.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+from griptape.drivers import BaseVectorStoreDriver, MongoDbAtlasVectorStoreDriver
+from griptape.utils import import_optional_dependency
+
+if TYPE_CHECKING:
+    from pymongo import MongoClient
+    from pymongo.collection import Collection
+
+
+@define
+class AzureMongoDbvCoreVectorStoreDriver(MongoDbAtlasVectorStoreDriver):
```
Comment (dylanholmes) [inline griptape/drivers/vector/azure_mongodb_vector_store_driver.py:13]: How is this Azure specific?

Comment (vachillo) [reply griptape/drivers/vector/azure_mongodb_vector_store_driver.py:13]: the aggregation pipeline requires some specific `$search` parameters. 
```
  pipeline.append(
      {
          "$search": {
              "**cosmosSearch**": {
```

[the remark this answers is not in the document]
Comment (vachillo) [reply griptape/drivers/vector/azure_mongodb_vector_store_driver.py:13]: im not sure what the best name is for either one honestly. either way itll be super long.

[the remark this answers is not in the document]
Comment (collindutter) [reply griptape/drivers/vector/azure_mongodb_vector_store_driver.py:13]: It should match, yes. I think the other mongodb Driver is incorrectly named too.

Comment (collindutter) [inline griptape/drivers/vector/azure_mongodb_vector_store_driver.py:13]: AzureMongoDb**v**CoreVectorStoreDriver

Extra `v`?

**Review by collindutter — CHANGES_REQUESTED** [review]

File: griptape/drivers/vector/azure_mongodb_vector_store_driver.py (hunk 1 above)
Comment (vachillo) [reply griptape/drivers/vector/azure_mongodb_vector_store_driver.py:13]: the API is called `vCore`. any name suggestions?

Comment (vachillo) [reply griptape/drivers/vector/azure_mongodb_vector_store_driver.py]: updated to `AzureMongoDbVectorStoreDriver`

**Review by collindutter — APPROVED** [review]

**Review by collindutter — DISMISSED** [review]

**Review by collindutter — COMMENTED** [review]
Looks good, can you please update CHANGELOG.md?

**Review by collindutter — APPROVED** [review]

**Review by collindutter — APPROVED** [review]

**Pull request merged**